public/src/main/java/com/revenuecat/purchases/models/PricingPhase.kt

```diff
@@ -28,4 +28,27 @@ data class PricingPhase(
      * [Price] of the [PricingPhase]
      */
     val price: Price
-) : Parcelable
+) : Parcelable {
+
+    /**
+     * Indicates how the pricing phase is charged for FINITE_RECURRING pricing phases
+     */
+    val paymentMode: PaymentMode?
+        get() {
+            // billingCycleCount is null for INFINITE_RECURRING or NON_RECURRING recurrence modes
+            // but validating for FINITE_RECURRING anyway
+            if (recurrenceMode != RecurrenceMode.FINITE_RECURRING) {
+                return null
+            }
```

Comment:
This might be the thing I'm unsure of the most 🤔 

Having a `PaymentMode` for `INFINITE_RECURRING` and `NON_RECURRING` doesn't _really_ add a lot of value since those modes are for recurring subscriptions and prepaid subscriptions. 

This mode is only for some complex logic for `FINITE_RECURRING`. With that... I'm not really sure if the name of `paymentMode` should be renamed to something like more specifc? 🤷‍♂️ 

Happy to hear any thoughts!

Reply:
`OfferPaymentMode` maybe? I think it only applies to offers, right? 
Otherwise we could also cover `INFINITE_RECURRING` and `NON_RECURRING` and give them their own special values in `PaymentMode`...

Reply:
I'm wondering if we could use `PAY_UP_FRONT` for those... It will always have that value, but I think it would fit in how those models work? Not sure though

Reply:
I _think_ I like the rename to `OfferPaymentMode` 🤔 That feels a lot better with what this is actually built. That leaves the full price phases without have any `offerPaymentMode` which makes them kind of stand out a bit nicer and not get lost in the offer pricing phases 🤷‍♂️